sdks/java/extensions/euphoria/src/main/java/org/apache/beam/sdk/extensions/euphoria/core/client/operator/FlatMap.java

```diff
@@ -153,7 +154,21 @@ public static OfBuilder named(@Nullable String name) {
      * @param eventTimeFn the event time extraction function
      * @return the next builder to complete the setup of the {@link FlatMap} operator
      */
-    Builders.Output<OutputT> eventTimeBy(ExtractEventTime<InputT> eventTimeFn);
+    default Builders.Output<OutputT> eventTimeBy(ExtractEventTime<InputT> eventTimeFn) {
+      // allowed timestamp shifts to infitive past
+      return eventTimeBy(eventTimeFn, Long.MAX_VALUE);
```

Comment:
Is this a reasonable default? Afaik Java SDK defaults to 0.

Reply:
The default favors functionality. We cannot know if the assignment function moves timestamps forward or backward, so this default ensures that any assignment will work. Moreover (if I understand it correctly) it has no performance or any other penalty.